tf2/include/tf2/buffer_core.h

```diff
@@ -317,19 +326,24 @@ class BufferCore
   
   /** \brief The pointers to potential frames that the tree can be made of.
    * The frames will be dynamically allocated at run time when set the first time. */
-  typedef std::vector<TimeCacheInterfacePtr> V_TimeCacheInterface;
-  V_TimeCacheInterface frames_;
-  
+  typedef std::map<CompactFrameID, TimeCacheInterfacePtr> M_TimeCacheInterface;
+  M_TimeCacheInterface frames_;
+
   /** \brief A mutex to protect testing and allocating new frames on the above vector. */
   mutable boost::mutex frame_mutex_;
 
   /** \brief A map from string frame ids to CompactFrameID */
   typedef boost::unordered_map<std::string, CompactFrameID> M_StringToCompactFrameID;
   M_StringToCompactFrameID frameIDs_;
   /** \brief A map from CompactFrameID frame_id_numbers to string for debugging and output */
-  std::vector<std::string> frameIDs_reverse;
+  typedef std::map<CompactFrameID, std::string> M_CompactFrameIDToString;
```

Comment:
If you don't really need to iterate over the elements of the frameIDs_reverse map, maybe an unordered_map would be enough?